tests/demo/test_demo_forward.py

```diff
@@ -0,0 +1,95 @@
+import pytest as pytest
+
+from didcomm.did_doc.did_resolver import register_default_did_resolver, DIDResolverChain
+from didcomm.pack import pack
+from didcomm.plaintext import Plaintext, PlaintextOptionalHeaders
+from didcomm.protocols.forward.forward import unpack_forward, wrap_in_forward
+from didcomm.secrets.secrets_resolver import register_default_secrets_resolver
+from didcomm.unpack import unpack, UnpackConfig
+from tests.common.interfaces_test import TestSecretsResolver, TestDIDResolver
+
+ALICE_DID = "did:example:alice"
+BOB_DID = "did:example:bob"
+
+
+@pytest.mark.asyncio
+async def test_demo_forward():
+    register_default_did_resolver(
+        DIDResolverChain([TestDIDResolver()])
+    )
+    register_default_secrets_resolver(TestSecretsResolver())
+
+    # ALICE
+    plaintext = Plaintext(body={"aaa": 1, "bbb": 2},
+                          id="1234567890", type="my-protocol/1.0",
+                          frm=ALICE_DID, to=[BOB_DID],
+                          created_time=1516269022, expires_time=1516385931)
+    pack_result = await pack(plaintext=plaintext, frm=ALICE_DID, to=BOB_DID)
+    print(f"Sending ${pack_result.packed_msg} to ${pack_result.service_endpoint}")
+
+    # BOB MEDIATOR
+    forward_bob = await unpack_forward(packed_msg=pack_result.packed_msg)
```

Comment:
Is a forward message created implicitly from Bob's DID doc above? It looks like Plaintext and I don't see `wrap_in_forward`

Reply:
When compared to `test_demo_mediators_unknown_to_sender`, where I can see that happening

Reply:
Yes, for the end user the output of the `pack` looks the same regardless of if the message is wrapped in Forward or it's a Plaintext. And this is done so intentionally:  We hide these details from the user, and doesn't require the user to understand (or even be aware of) Forward protocol. We just return a packed message which needs to be sent to the given endpoint. 

The packed message can be either Forward message (if Forwarding needs to be done) or a plaintext message.

`pack` method can handle it automatically resolving Bob's DID and analysing serviceEnpoint.

This auto-wrapping in Forward is enabled by default (PackConfig.forward=True) but can be disabled for advanced users. 

`wrap_in_forward` is a helper method that will be used by `pack` under the hood. We are considering make it public as it will be needed by Mediators to do re-wrapping. But in any case, this is part of 'advanced' API, and not needed for most of the users.